src/pages/reports/comparativeReport/AddEmployee.tsx

```diff
@@ -0,0 +1,121 @@
+import React, {useState} from 'react'
+import {Box, Button, FormControl, IconButton, Paper, TextField, Typography} from '@mui/material'
+import useCreateEventStyles from '../../../assets/styleJs/events/createEvent'
+import {v4 as uuidv4} from 'uuid'
+import DeleteIcon from '../../../assets/images/Icons/DeleteIcon'
+import useReportsStyle from '../../../assets/styleJs/report/report'
+import {randomColor} from '../../../utils/utils'
+import {IEmployee} from '../../../types/eployeeTypes'
+
+const AddEmployee = () => {
+  const classesCreateEvent = useCreateEventStyles()
+  const classes = useReportsStyle()
+  const [employee, setEmployee] = useState<IEmployee[]>([
+    {
+      id: uuidv4(),
+      name: '',
+      position: '',
+      employee: 'Employee 1',
+      bgColor: randomColor(),
+    },
+    {
+      id: uuidv4(),
+      name: '',
+      position: '',
+      employee: 'Employee 2',
+      bgColor: randomColor(),
+    },
+  ])
+
+  const handleAddEmployee = () => {
+    const tempArr = []
+    const employeeNumber = employee.length + 1
+    tempArr[0] = {
+      id: uuidv4(),
+      name: '',
+      position: '',
+      employee: 'Employee ' + employeeNumber,
+      bgColor: randomColor(),
+    }
+    setEmployee([...employee, ...tempArr])
+  }
+  const onChangeEmployeeData = (id: string, value: string, changingValue: string) => {
+    const newArray = employee.map((item: IEmployee) => {
+      if (item.id === id) {
+        return {...item, [changingValue]: value}
+      }
+      return item
+    })
+    setEmployee(newArray)
+  }
+  const handleRemoveEmployee = (item: IEmployee) => {
+    const newList = [...employee]
+    const index = employee.indexOf(item)
+    if (index > -1) {
+      newList.splice(index, 1)
+    }
+    setEmployee(newList)
```

Comment:
Can we avoid using functions that are modifying source variables, here we used the splice function? We can achieve the same result by using a filter